app/components/Form/ProjectManagerForm.tsx

```diff
@@ -1,64 +1,262 @@
+        newFormData: data,
+      },
+      optimisticResponse: {
+        createFormChange: {
+          query: {
+            projectRevision: {
+              managerFormChanges: {
+                edges: {
+                  change: {
+                    projectManagerLabel: {},
+                    formChange: {
+                      id: "new",
+                      newFormData: data,
+                    },
+                  },
+                },
+              },
+            },
+          },
+        },
+      },
+    });
+  };
+
+  const [applyUpdateFormChangeMutation] = useDebouncedMutation(
+    updateFormChangeMutation
+  );
+
+  // Delete a manager from the project revision
+  const [discardFormChange, discardInFlight] =
+    useDeleteManagerFromRevisionMutation();
+  const deleteManager = (id: string) => {
+    if (change.formChange.operation === "CREATE")
+      discardFormChange({
+        variables: {
+          input: {
+            id: id,
+          },
+          projectRevision: projectRevisionId,
+        },
+        onError: (error) => {
+          console.log(error);
+        },
+      });
+    else
+      applyUpdateFormChangeMutation({
+        variables: {
+          input: {
+            id: id,
+            formChangePatch: {
+              operation: "ARCHIVE",
+            },
+          },
+        },
+        optimisticResponse: {
+          updateFormChange: {
+            formChange: {
+              id: id,
+              newFormData: {},
+            },
+          },
+        },
+        onError: (error) => {
+          console.log(error);
+        },
+        debounceKey: id,
+      });
+  };
+
+  // Update an existing project_manager form change if it exists, otherwise create one
+  const createOrUpdateFormChange = (
+    formChangeId: string,
+    labelId: number,
+    formChange: { cifUserId: number }
+  ) => {
+    const data = {
+      ...formChange,
+      projectManagerLabelId: labelId,
+      projectId: projectId,
+    };
+
+    // If a form_change already exists, and the payload contains a cifUserId update it
+    if (formChangeId && formChange?.cifUserId) {
+      applyUpdateFormChangeMutation({
+        variables: {
+          input: {
+            id: formChangeId,
+            formChangePatch: {
+              newFormData: data,
+            },
+          },
+        },
+        optimisticResponse: {
+          updateFormChange: {
+            formChange: {
+              id: formChangeId,
+              newFormData: data,
+            },
+          },
+        },
+        debounceKey: formChangeId,
+      });
+      // If a form_change does not exist, and the payload contains a cifUserId create a form_change record
+    } else if (formChange?.cifUserId) {
+      addManager(data);
+    }
+    // If a form_change exists, and the payload does not contain a cifUserId delete it
+    else if (formChangeId && !formChange.cifUserId && !discardInFlight) {
+      deleteManager(formChangeId);
+    }
```

Comment:
I feel this entire pattern of creating / updating / deleting form changes depending on whether we have a certain field or not will be reused over and over, just a noting a thought for later on how we could reuse that logic

Reply:
Yes, this may be refactored into a hook. We can do that after we have two examples of this pattern (there will be a second one in #280 )